autofaiss/external/optimize.py

```diff
@@ -154,25 +156,54 @@ def get_optimal_index_keys_v2(
     """
 
     # Exception cases:
-    if nb_vectors < flat_threshold:  # HNSW Faiss slower when less than 1000 vectors
-        return ["Flat"]
-    if nb_vectors < hnsw_threshold:  # HNSW is faster and quantization is not possible (>=10_000 vectors needed)
+
+    if nb_vectors < flat_threshold:  # When less than 1000 vectors, the flat index is usually faster
+        return ["IVF1,Flat" if should_be_memory_mappable else "Flat"]
+    if nb_vectors < quantization_threshold:  # quantization is not possible (>=10_000 vectors needed)
+        if should_be_memory_mappable:
+            return get_optimal_ivf(nb_vectors)
         return ["HNSW15"]
-    if force_pq is not None:
+    if force_pq is not None:  # Forced quantization
         return get_optimal_quantization(nb_vectors, dim_vector, force_quantization_value=force_pq)
 
+    # General cases:
+
     # Get max memory usage
     max_size_in_bytes = cast_memory_to_bytes(max_index_memory_usage)
 
-    # If we can build an HNSW with the given memory constraints, it's the best
-    m_hnsw = int(floor((max_size_in_bytes / (4 * nb_vectors) - dim_vector) / 2))
-    if m_hnsw >= 8:
-        return [f"HNSW{min(m_hnsw, 32)}"]
+    if not should_be_memory_mappable:
+        # If we can build an HNSW with the given memory constraints, it's the best
+        m_hnsw = int(floor((max_size_in_bytes / (4 * nb_vectors) - dim_vector) / 2))
+        if m_hnsw >= 8:
+            return [f"HNSW{min(m_hnsw, 32)}"]
+
+    # Try to build a not quantized IVF index
+    index_keys = get_optimal_ivf(nb_vectors)
+    index_metadata = IndexMetadata(index_keys[0], nb_vectors, dim_vector, make_direct_map=False)
```

Comment:
why make direct map hard coded to false ?

Reply:
Just because this function doesn't take make_direct_map as parameter
It is free to add it so let's do it